daml-lf/transaction/src/test/scala/com/digitalasset/daml/lf/transaction/ContractStateMachineSpec.scala

```diff
@@ -134,6 +134,12 @@ class ContractStateMachineSpec extends AnyWordSpec with Matchers with TableDrive
       version = txVersion,
     )
 
+  implicit def inconsistentContractKey[X](key: GlobalKey): Left[KeyInputError, X] =
+    Left(Left(InconsistentContractKey(key)))
+
+  implicit def duplicateContractKey[X](key: GlobalKey): Left[KeyInputError, X] =
+    Left(Right(DuplicateContractKey(key)))
```

Comment:
I don't think the `implicit` makes any sense here because Scala cannot on its own decide which of the two implicits to use as the types are the same.
```suggestion
  def inconsistentContractKey[X](key: GlobalKey): Left[KeyInputError, X] =
    Left(Left(InconsistentContractKey(key)))

  def duplicateContractKey[X](key: GlobalKey): Left[KeyInputError, X] =
    Left(Right(DuplicateContractKey(key)))
```